src/client/app/dashboard/patient-dashboard.component.ts

```diff
@@ -39,52 +43,84 @@ export class PatientDashboardComponent implements OnInit {
 
     ngOnInit() {
         this.navbarComponent.navbarColor(0, '#6960FF');
-        this.chart();
         this.visitorId = +this.route.snapshot.paramMap.get('id');// this will give the visitor id
         const cookie = this.securityService.getCookie('userDetails');
         if (cookie === '') {
             this.router.navigate([`/login`]);
-        } else if (this.visitorId === JSON.parse(cookie).id) {
+        } else {
             this.chatService.getUserById(this.visitorId)
                 .subscribe(user => {
                     this.selectedUser = user;
+                    if (this.selectedUser.picUrl) {
+                        this.downloadPic(this.selectedUser.picUrl);
+                    } else {
+                        this.downloadAltPic(this.selectedUser.role);
+                    }
                 });
             this.getVisitor(this.visitorId);
             this.getVisitorStore(this.visitorId);
-            this.getVisitorAppointment(this.visitorId);
-            this.visitorAppointment = {'status': 'online'};
+            this.getVisitorAppointmentHistory(this.visitorId);
             this.getVisitorReport(this.visitorId);
             this.getVisitorHealth(this.visitorId);
             this.getVisitorPrescription(this.visitorId);
+            this.getTimeline(this.visitorId);
         }
     }
 
-    chart() {
+    downloadPic(filename: string) {
+        this.chatService.downloadFile(filename)
+            .subscribe((res: any) => {
+                res.onloadend = () => {
+                    this.picUrl = this.domSanitizer.bypassSecurityTrustUrl(res.result);
```

Comment:
You may as well use the `safe pipe` which sanitizes the url instead of doing this every time in every function that imports a new url.